docs/Clojure.md

```diff
@@ -0,0 +1,96 @@
+# How to use Minio SDK for java with Clojure
+
+In this receipe we will learn how to use `minio-java` with Clojure.
+
+## Prerequisites
+Install [`Clojure`](https://clojure.org/community/downloads) and [`Leiningen`](https://leiningen.org/).
+
+Check by running -
+```
+lein --help
+```
+
+## Steps for setup
+Create a new Clojure project using Leiningen.
+```
+lein new <templateName> <projectName>
+```
+
+```
+lein new app clojproject
+```
+
+Navigate to `project.clj` look at `:dependencies` section, listing what project requires in order to load, compile and run.
+```
+lein deps
+```
+Run this command to install list of dependencies to your project.
+
+Use Leiningen to start Clojure now - 
+```
+lein repl
+```
+
+Test the project by running `(-main)`.
+```
+user=> (-main)
+Hello, World!
+nil
+```
+
+Or you can run the project using command - `lein run`.
+
+## Getting started with minio-java
+
+### Download minio-java from maven
+Go to [`maven`](http://search.maven.org/) and search `minio`. 
+Download the lastest version and add dependency in the `project.clj`.
+
+```
+::dependencies [
+                [io.minio/minio "4.0.2"]
+               ]
+```
+Run `lein deps`.
+
+Check `lein classpath` to see if minio.jar is present.
+```
+/.m2/repository/io/minio/minio/4.0.2/minio-4.0.2.jar
+```
+
+Relaunch repl for Clojure and use minio-java.
+
+## Test
+Use classes and methods from the imported class in your project.
+
+### Make Bucket
+```
+(ns clojproject.core
+ (import io.minio.MinioClient)
+)
+
+(defn -main
+  "I don't do a whole lot ... yet."
+  [& args]
+  (println "Hello, World!")
+    (println "Using minio-java")
+    (.makeBucket (new io.minio.MinioClient "https://play.minio.io:9000" "Q3AM3UQ867SPQQA43P2F" "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG") "testbucketcloj")
+    )
+    
+```
+
+### PutObject
+```
+(ns clojproject.core
+ (import io.minio.MinioClient)
+)
+
+(defn -main
+  "I don't do a whole lot ... yet."
```

Comment:
We don't need to write custom sentences like these, we should write what the `main` function does.